### DISCUSSION TITLE
How to install old version of hoomd from tarball

### DISCUSSION THREAD

**Discussion by jmarcum2** [glotzerlab/hoomd-blue#1760] [Support] [answered]
### Overview
I need to install hoomd 3.8.0. It seems like I'll have to use the 3.8.0 tarball from [the releases page](https://github.com/glotzerlab/hoomd-blue/releases?page=2). I do not have any experience installing packages this way and I am struggling to install this tarball.

### Further context
I have a project I have been working on for a year that depends on hoomd 3.8.0. Both my computer and my profile on my university's computing cluster have been using this version of hoomd. When I logged on to my computer today, I found that my conda environment is using hoomd version 2.9.7 (??) and my scripts of course don't recognize many of the functions and methods that they should be using. (Computing cluster still has 3.8.0). Regardless of how this happened, I need to install hoomd 3.8.0, which seems to be no longer available directly from anaconda. If I can figure this out, it would be much more ideal than updating my project to support the latest version of hoomd and ensure that all of the behavior is correct.

### What I have tried 
I have been following this stackoverflow question on [Installing modules to anaconda from .tar.gz](https://stackoverflow.com/questions/45156080/installing-modules-to-anaconda-from-tar-gz/45167575#45167575), changing things as necessary to apply to my MacOS system. 

My conda environment path is `/opt/anaconda3/myenv` and when I activate this environment and use `which python` to find my python path, it returns `/opt/anaconda3/envs/myenv/bin/python`. So, following the stackoverflow answer, I have tried:
`/opt/anaconda3/envs/myenv/bin/python -m pip install ~/Downloads/hoomd-v3.8.0.tar.gz `
resulting in the error:
`ERROR: file:///Users/[myname]/Downloads/hoomd-v3.8.0.tar.gz does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.`

Indeed, these files are not contained within the tarball.

I have also tried installing to a different path. When I open python and query `hoomd.__file__` I obtain the path: `/opt/anaconda3/envs/myenv/lib/python3.10/site-packages/hoomd/__init__.py`. 

Am I at all on the right track with this last step? When I try to install the tarball on any level of this new structure starting at `/python3.10` Ι am denied permission: 
`zsh: permission denied: /opt/anaconda3/envs/myenv/lib/python3.10`.
When I "get info" on this folder in Finder (not using command line) it says I have permissions to read and write.

### Final notes:
There are two differently-named tarballs for hoomd 3.8.0 on the release page and I have tried both of them. 

Am I on the right track at all? Has anyone else faced this issue, or can anyone shed light on the proper way to install an old version of hoomd from a tarball?

Am I missing another way to install the old version? `conda install hoomd=3.8.0` does not work.

**Comment by chrisjonesBSU** [comment]
Have you tried installing hoomd 3.8 from anaconda?

Try it out, make a new anaconda environment called `hoomd3` and ask conda to install hoomd v 3.8 (from the conda-forge channel) into it.

```
conda create n hoomd3 -c conda-forge hoomd=3.8
```

Anaconda will list a ton of packages it wants to install into this environment, scroll through it until you see `hoomd` and you can see which version it is pulling.

**Comment by chrisjonesBSU** [comment, marked as answer]
Oops, I just now read your last sentence. Make sure you're instructing anaconda to use the conda-forge channel, so try your command again using the `-c` flag `-c conda-forge`

**Reply by jmarcum2** [reply]
Thank you Chris! This worked

**Reply by joaander** [reply]
Package incompatibilities can lead to problems installing new software in old environments. To avoid these, use `<package-manager> update --all` to update all your packages and follow the instructions on `conda-forge` to add the `conda-forge` channel as default: https://conda-forge.org/docs/user/introduction/.  I recommend removing the `default` channel entirely. 

Also, why do you need 3.8? We have fixed dozens of bugs and added many new features between 3.8 and 4.6.
